### PR TITLE
[cms] DCC v1

### LINKED ISSUE #1537: [cms] DCC listings
So just like proposals or invoices, there needs to be a page that shows a list of DCC proposals that are currently in various categories (Approved, Rejected, Active etc)

All DCC's themselves can be requested via the `/dcc` route with an optional status.

### PR DESCRIPTION
**Pull request by victorgcramos**
This diff adds the DCC module based on CMS v1, including:

1. Submit DCC #1539 
2. DCC listings #1537 
3. DCC Details #1538 
4. DCC admin Approve/reject #1541 
5. DCC Support/Oppose #1540 

Closes #1537, #1538, #1539, #1540, #1541

### CODE REVIEW

File: src/actions/app.js [hunk 1]
```diff
@@ -438,3 +502,34 @@ export const onResetComments = () => act.RESET_COMMENTS();
...
+    return;
+  }
+
+  let dcc = null;
+  for (const status in fetchedDCCsByStatus) {
+    const dccFromStatus = fetchedDCCsByStatus[status].find(
+      d => d.censorshiprecord.token === token
+    );
+    dcc = dccFromStatus ? dccFromStatus : dcc;
+  }
```
Comment (tiagoalvesdulce) [inline src/actions/app.js]: You can import `flow`, `values`, `flattenDeep` and `find` from `lodash/fp` and rewrite this code in a more elegant and readable way. See:

```
const dcc = flow(
    values,
    flattenDeep,
    find(dcc => dcc.censorshiprecord.token === token)
  )(fetchedDCCsByStatus)
```
Always prefer a functional approach.

File: src/components/DCC/List/Card.jsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  nomineeusername,
+  status
+}) => {
+  const { type, domain, contractortype } = dccpayload;
+
+  return (
+    <div className={`thing thing-proposal id-${token} odd link`}>
+      <span
+        className="title"
+        style={{ display: "flex", overflow: "visible" }}
```
Comment (tiagoalvesdulce) [inline src/components/DCC/List/Card.jsx]: I know it's simple styling but let's use CSS files for it.

File: src/components/DCC/List/List.jsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+          <div style={{ float: "right" }}>
+            <Tab
+              title="drafts"
+              selected={isDraftOption}
+              onTabChange={() => handleStatusChange(DCC_STATUS_DRAFTS)}
+            />
+          </div>
+        </Tabs>
+        {isDraftOption ?
+          <ListDraft drafts={drafts || {}} onDeleteDraft={id => e => {
```
Comment (tiagoalvesdulce) [inline src/components/DCC/List/List.jsx]: This should be broken into one or two methods

File: src/components/DCC/List/hooks.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    },
+    [onDeleteDraftDCC]
+  );
+
+  useEffect(
+    function onChangeStatus() {
+      async function onFetchData() {
+        setLoadingDCCs(true);
+        await onFetchDCCsByStatus(status);
+        // force timeout
```
Comment (tiagoalvesdulce) [inline src/components/DCC/List/hooks.js]: Why do you need this here?

File: src/components/DCC/New/hooks.js [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+      const id = onSaveDraftDCC({
+        type: dcctype,
+        nomineeid: nomineeid,
+        statement: dccstatement,
+        domain: dccdomain,
+        contractortype: contractortype,
+        draftId
+      });
+      setFakeLoadingDraft(true);
+      setTimeout(() => {
```
Comment (tiagoalvesdulce) [inline src/components/DCC/New/hooks.js]: Another timeout. Don't we have another way of doing it?

File: src/components/DCC/New/hooks.js [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+      setSavedDraft,
+      setFakeLoadingDraft,
+      setDraftId,
+      draftId
+    ]
+  );
+
+  useEffect(() => {
+    if (requestDone && !error && newToken) {
+      setTimeout(() => {
```
Comment (tiagoalvesdulce) [inline src/components/DCC/New/hooks.js]: another one

File: src/actions/app.js (hunk 1 above)
Comment (victorgcramos) [reply src/actions/app.js]: Thanks for the advice, @tiagoalvesdulce!! #TMJ

**Comment by alexlyp** [comment]
Overall looks pretty solid.

Couple of things I've noticed:
1) After a successfully approved revocation, the user details page is inaccessible and the following error is provided: 

TypeError: user.identities[(user.identities.length - 1)] is undefined
src/github.com/decred/politeiagui/src/components/UserDetail/GeneralTab.js:264

May actually be happening on master as well, so guessing it could be unrelated to the DCC work.

2) I have a DCC in the Active listing that has a Title simply as "0", upon clicking on it, it gives the following error:  

TypeError: type is undefined
src/github.com/decred/politeiagui/src/components/DCC/Details/Details.jsx:88

**Comment by alexlyp** [comment]
So just verified the cause of the first issue above is with trying to access user details for users that have not yet fully created their account.  Still should probably be fixed to avoid the error, and instead just show a message pointing out the user's current status.

**Pull request merged**